### DISCUSSION TITLE
Migration from Old Version

### DISCUSSION THREAD

**Discussion by pts-jeffreyjewitt** [netbox-community/netbox-docker#995] [Getting Help]
### Current Behavior

Trying to upgrade to version 3.4.8 from a version (unknown 2.x.x).
After importing the DB, getting this error:
netbox-docker-netbox-1               | ⏳ Waiting on DB... (3s / 30s)
netbox-docker-netbox-1               | django.db.migrations.exceptions.NodeNotFoundError: Migration dcim.0131_consoleport_speed depends on nonexistent node ('dcim', '0130_sitegroup'). Django tried to replace migration dcim.0130_sitegroup with any of [dcim.0003_squashed_0130] but wasn't able to because some of the replaced migrations are already applied.
netbox-docker-netbox-1               | [ Use DB_WAIT_DEBUG=1 in netbox.env to print full traceback for errors here ]
netbox-docker-netbox-1               | ⏳ Waiting on DB... (6s / 30s)
netbox-docker-netbox-1               | django.db.migrations.exceptions.NodeNotFoundError: Migration dcim.0131_consoleport_speed depends on nonexistent node ('dcim', '0130_sitegroup'). Django tried to replace migration dcim.0130_sitegroup with any of [dcim.0003_squashed_0130] but wasn't able to because some of the replaced migrations are already applied.
netbox-docker-netbox-1               | [ Use DB_WAIT_DEBUG=1 in netbox.env to print full traceback for errors here ]
netbox-docker-netbox-1               | ⏳ Waiting on DB... (9s / 30s)
netbox-docker-netbox-1               | django.db.migrations.exceptions.NodeNotFoundError: Migration dcim.0131_consoleport_speed depends on nonexistent node ('dcim', '0130_sitegroup'). Django tried to replace migration dcim.0130_sitegroup with any of [dcim.0003_squashed_0130] but wasn't able to because some of the replaced migrations are already applied.
netbox-docker-netbox-1               | [ Use DB_WAIT_DEBUG=1 in netbox.env to print full traceback for errors here ]
netbox-docker-netbox-1               | ⏳ Waiting on DB... (12s / 30s)

Is there a way to force the DB migrations to complete?
Thanks

### Expected Behavior

Import data, that it takes the data and migrates it

### Docker Compose Version

Docker Compose version v2.3.3

### Docker Version

```text
Client: Docker Engine - Community
 Version:           23.0.4
 API version:       1.42
 Go version:        go1.19.8
 Git commit:        f480fb1
 Built:             Fri Apr 14 10:32:04 2023
 OS/Arch:           linux/amd64
 Context:           default

Server: Docker Engine - Community
 Engine:
  Version:          23.0.4
  API version:      1.42 (minimum version 1.12)
  Go version:       go1.19.8
  Git commit:       cbce331
  Built:            Fri Apr 14 10:32:04 2023
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.20
  GitCommit:        2806fc1057397dbaeefbea0e4e17bddfbd388f38
 runc:
  Version:          1.1.5
  GitCommit:        v1.1.5-0-gf19387a
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

### The git Revision

61a3afbb3b83b294bf87d6d34e78b318f03476da

### The git Status

```text
On branch release
Your branch is up to date with 'origin/release'.
```

### Startup Command

docker compose up / docker compose start

### NetBox Logs

```text
Attaching to netbox-docker-netbox-1, netbox-docker-netbox-housekeeping-1, netbox-docker-netbox-worker-1, netbox-docker-postgres-1, netbox-docker-redis-1, netbox-docker-redis-cache-1, netbox-docker-tls-1
netbox-docker-redis-1                | 1:C 24 Apr 2023 16:58:34.658 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
netbox-docker-redis-1                | 1:C 24 Apr 2023 16:58:34.660 # Redis version=7.0.11, bits=64, commit=00000000, modified=0, pid=1, just started
netbox-docker-redis-1                | 1:C 24 Apr 2023 16:58:34.660 # Configuration loaded
netbox-docker-redis-1                | 1:M 24 Apr 2023 16:58:34.665 * monotonic clock: POSIX clock_gettime
netbox-docker-redis-1                | 1:M 24 Apr 2023 16:58:34.672 * Running mode=standalone, port=6379.
netbox-docker-redis-1                | 1:M 24 Apr 2023 16:58:34.672 # Server initialized
netbox-docker-redis-1                | 1:M 24 Apr 2023 16:58:34.672 # WARNING Memory overcommit must be enabled! Without it, a background save or replication may fail under low memory condition. Being disabled, it can can also cause failures without low memory condition, see https://github.com/jemalloc/jemalloc/issues/1328. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
netbox-docker-redis-1                | 1:M 24 Apr 2023 16:58:34.673 * Reading RDB base file on AOF loading...
netbox-docker-redis-1                | 1:M 24 Apr 2023 16:58:34.673 * Loading RDB produced by version 7.0.11
netbox-docker-redis-1                | 1:M 24 Apr 2023 16:58:34.673 * RDB age 14800 seconds
netbox-docker-redis-1                | 1:M 24 Apr 2023 16:58:34.673 * RDB memory usage when created 0.82 Mb
netbox-docker-redis-1                | 1:M 24 Apr 2023 16:58:34.673 * RDB is base AOF
netbox-docker-redis-1                | 1:M 24 Apr 2023 16:58:34.675 * Done loading RDB, keys loaded: 0, keys expired: 0.
netbox-docker-redis-1                | 1:M 24 Apr 2023 16:58:34.675 * DB loaded from base file appendonly.aof.1.base.rdb: 0.002 seconds
netbox-docker-redis-1                | 1:M 24 Apr 2023 16:58:34.964 * DB loaded from incr file appendonly.aof.1.incr.aof: 0.289 seconds
netbox-docker-redis-1                | 1:M 24 Apr 2023 16:58:34.964 * DB loaded from append only file: 0.291 seconds
netbox-docker-redis-1                | 1:M 24 Apr 2023 16:58:34.964 * Opening AOF incr file appendonly.aof.1.incr.aof on server start
netbox-docker-redis-1                | 1:M 24 Apr 2023 16:58:34.964 * Ready to accept connections
netbox-docker-redis-cache-1          | 1:C 24 Apr 2023 16:58:35.074 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
netbox-docker-redis-cache-1          | 1:C 24 Apr 2023 16:58:35.074 # Redis version=7.0.11, bits=64, commit=00000000, modified=0, pid=1, just started
netbox-docker-redis-cache-1          | 1:C 24 Apr 2023 16:58:35.074 # Configuration loaded
netbox-docker-redis-cache-1          | 1:M 24 Apr 2023 16:58:35.075 * monotonic clock: POSIX clock_gettime
netbox-docker-redis-cache-1          | 1:M 24 Apr 2023 16:58:35.076 * Running mode=standalone, port=6379.
netbox-docker-redis-cache-1          | 1:M 24 Apr 2023 16:58:35.088 # Server initialized
netbox-docker-redis-cache-1          | 1:M 24 Apr 2023 16:58:35.088 # WARNING Memory overcommit must be enabled! Without it, a background save or replication may fail under low memory condition. Being disabled, it can can also cause failures without low memory condition, see https://github.com/jemalloc/jemalloc/issues/1328. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
netbox-docker-redis-cache-1          | 1:M 24 Apr 2023 16:58:35.109 * Loading RDB produced by version 7.0.11
netbox-docker-redis-cache-1          | 1:M 24 Apr 2023 16:58:35.109 * RDB age 164 seconds
netbox-docker-redis-cache-1          | 1:M 24 Apr 2023 16:58:35.109 * RDB memory usage when created 0.82 Mb
netbox-docker-redis-cache-1          | 1:M 24 Apr 2023 16:58:35.109 * Done loading RDB, keys loaded: 1, keys expired: 0.
netbox-docker-redis-cache-1          | 1:M 24 Apr 2023 16:58:35.109 * DB loaded from disk: 0.001 seconds
netbox-docker-redis-cache-1          | 1:M 24 Apr 2023 16:58:35.109 * Ready to accept connections
netbox-docker-postgres-1             | 
netbox-docker-postgres-1             | PostgreSQL Database directory appears to contain a database; Skipping initialization
netbox-docker-postgres-1             | 
netbox-docker-postgres-1             | 2023-04-24 16:58:35.378 UTC [1] LOG:  starting PostgreSQL 15.2 on x86_64-pc-linux-musl, compiled by gcc (Alpine 12.2.1_git20220924-r4) 12.2.1 20220924, 64-bit
netbox-docker-postgres-1             | 2023-04-24 16:58:35.382 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
netbox-docker-postgres-1             | 2023-04-24 16:58:35.383 UTC [1] LOG:  listening on IPv6 address "::", port 5432
netbox-docker-postgres-1             | 2023-04-24 16:58:35.393 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
netbox-docker-postgres-1             | 2023-04-24 16:58:35.429 UTC [24] LOG:  database system was shut down at 2023-04-24 16:58:16 UTC
netbox-docker-postgres-1             | 2023-04-24 16:58:35.535 UTC [1] LOG:  database system is ready to accept connections
netbox-docker-tls-1                  | {"level":"info","ts":1682355520.0201802,"msg":"using provided configuration","config_file":"/etc/caddy/Caddyfile","config_adapter":"caddyfile"}
netbox-docker-tls-1                  | {"level":"warn","ts":1682355520.0223038,"msg":"Caddyfile input is not formatted; run the 'caddy fmt' command to fix inconsistencies","adapter":"caddyfile","file":"/etc/caddy/Caddyfile","line":2}
netbox-docker-tls-1                  | {"level":"info","ts":1682355520.0291991,"logger":"admin","msg":"admin endpoint started","address":"localhost:2019","enforce_origin":false,"origins":["//localhost:2019","//[::1]:2019","//127.0.0.1:2019"]}
netbox-docker-tls-1                  | {"level":"info","ts":1682355520.033512,"logger":"tls.cache.maintenance","msg":"started background certificate maintenance","cache":"0xc0002c9f80"}
netbox-docker-tls-1                  | {"level":"info","ts":1682355520.096038,"logger":"http","msg":"skipping automatic certificate management because one or more matching certificates are already loaded","domain":"netbox.fqdn","server_name":"srv0"}
netbox-docker-tls-1                  | {"level":"info","ts":1682355520.0961032,"logger":"http","msg":"enabling automatic HTTP->HTTPS redirects","server_name":"srv0"}
netbox-docker-tls-1                  | {"level":"info","ts":1682355520.0980558,"logger":"http","msg":"enabling HTTP/3 listener","addr":":443"}
netbox-docker-tls-1                  | {"level":"info","ts":1682355520.0988243,"msg":"failed to sufficiently increase receive buffer size (was: 208 kiB, wanted: 2048 kiB, got: 416 kiB). See https://github.com/quic-go/quic-go/wiki/UDP-Receive-Buffer-Size for details."}
netbox-docker-tls-1                  | {"level":"info","ts":1682355520.0994244,"logger":"http.log","msg":"server running","name":"srv0","protocols":["h1","h2","h3"]}
netbox-docker-tls-1                  | {"level":"info","ts":1682355520.102166,"logger":"tls","msg":"cleaning storage unit","description":"FileStorage:/data/caddy"}
netbox-docker-tls-1                  | {"level":"info","ts":1682355520.104266,"logger":"tls","msg":"finished cleaning storage units"}
netbox-docker-tls-1                  | {"level":"info","ts":1682355520.1066184,"logger":"http.log","msg":"server running","name":"remaining_auto_https_redirects","protocols":["h1","h2","h3"]}
netbox-docker-tls-1                  | {"level":"info","ts":1682355520.1104987,"msg":"autosaved config (load with --resume flag)","file":"/config/caddy/autosave.json"}
netbox-docker-tls-1                  | {"level":"info","ts":1682355520.1105947,"msg":"serving initial configuration"}
netbox-docker-tls-1                  | {"level":"error","ts":1682355520.9195051,"logger":"http.log.error.log0","msg":"dial tcp 172.27.0.5:8080: connect: connection refused","request":{"remote_ip":"IPSTART.0.19","remote_port":"37078","proto":"HTTP/2.0","method":"GET","host":"netbox.fqdn","uri":"/","headers":{"Sec-Ch-Ua-Platform":["\"Linux\""],"Accept-Language":["en-US,en;q=0.9"],"Cookie":[],"Sec-Ch-Ua-Mobile":["?0"],"Upgrade-Insecure-Requests":["1"],"Accept-Encoding":["gzip, deflate, br"],"Cache-Control":["max-age=0"],"Sec-Fetch-User":["?1"],"Sec-Fetch-Dest":["document"],"Sec-Fetch-Mode":["navigate"],"Sec-Ch-Ua":["\"Google Chrome\";v=\"111\", \"Not(A:Brand\";v=\"8\", \"Chromium\";v=\"111\""],"User-Agent":["Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36"],"Accept":["text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7"],"Sec-Fetch-Site":["none"]},"tls":{"resumed":false,"version":772,"cipher_suite":4867,"proto":"h2","server_name":"netbox.fqdn"}},"duration":0.027349836,"status":502,"err_id":"95mi125hw","err_trace":"reverseproxy.statusError (reverseproxy.go:1299)"}
netbox-docker-tls-1                  | {"level":"error","ts":1682355520.919573,"logger":"http.log.access.log0","msg":"handled request","request":{"remote_ip":"IPSTART.0.19","remote_port":"37078","proto":"HTTP/2.0","method":"GE-stT","host":"netbox.fqdn","uri":"/","headers":{"Sec-Ch-Ua-Platform":["\"Linux\""],"Accept-Language":["en-US,en;q=0.9"],"Cookie":[],"Sec-Ch-Ua-Mobile":["?0"],"Upgrade-Insecure-Requests":["1"],"Accept-Encoding":["gzip, deflate, br"],"Cache-Control":["max-age=0"],"Sec-Fetch-User":["?1"],"Sec-Fetch-Dest":["document"],"Sec-Fetch-Mode":["navigate"],"Sec-Ch-Ua":["\"Google Chrome\";v=\"111\", \"Not(A:Brand\";v=\"8\", \"Chromium\";v=\"111\""],"User-Agent":["Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36"],"Accept":["text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7"],"Sec-Fetch-Site":["none"]},"tls":{"resumed":false,"version":772,"cipher_suite":4867,"proto":"h2","server_name":"netbox.fqdn"}},"user_id":"","duration":0.027349836,"size":0,"status":502,"resp_headers":{"Server":["Caddy"],"Alt-Svc":["h3=\":443\"; ma=2592000"]}}
netbox-docker-netbox-1               | django.db.migrations.exceptions.NodeNotFoundError: Migration dcim.0131_consoleport_speed depends on nonexistent node ('dcim', '0130_sitegroup'). Django tried to replace migration dcim.0130_sitegroup with any of [dcim.0003_squashed_0130] but wasn't able to because some of the replaced migrations are already applied.
netbox-docker-netbox-1               | [ Use DB_WAIT_DEBUG=1 in netbox.env to print full traceback for errors here ]
netbox-docker-netbox-1               | ⏳ Waiting on DB... (0s / 30s)
netbox-docker-tls-1                  | {"level":"error","ts":1682355539.1213171,"logger":"http.log.error.log0","msg":"dial tcp 172.27.0.5:8080: connect: connection refused","request":{"remote_ip":"IPSTART.0.19","remote_port":"37078","proto":"HTTP/2.0","method":"GET","host":"netbox.fqdn","uri":"/","headers":{"Sec-Fetch-Site":["none"],"Accept-Encoding":["gzip, deflate, br"],"Sec-Ch-Ua-Mobile":["?0"],"Sec-Ch-Ua":["\"Google Chrome\";v=\"111\", \"Not(A:Brand\";v=\"8\", \"Chromium\";v=\"111\""],"Sec-Ch-Ua-Platform":["\"Linux\""],"Accept":["text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7"],"Sec-Fetch-Mode":["navigate"],"Cache-Control":["max-age=0"],"Sec-Fetch-User":["?1"],"Accept-Language":["en-US,en;q=0.9"],"Upgrade-Insecure-Requests":["1"],"Sec-Fetch-Dest":["document"],"Cookie":[],"User-Agent":["Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36"]},"tls":{"resumed":false,"version":772,"cipher_suite":4867,"proto":"h2","server_name":"netbox.fqdn"}},"duration":0.00596638,"status":502,"err_id":"02exdc2pv","err_trace":"reverseproxy.statusError (reverseproxy.go:1299)"}
netbox-docker-tls-1                  | {"level":"error","ts":1682355539.1213913,"logger":"http.log.access.log0","msg":"handled request","request":{"remote_ip":"IPSTART.0.19","remote_port":"37078","proto":"HTTP/2.0","method":"GET","host":"netbox.fqdn","uri":"/","headers":{"Accept":["text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7"],"Sec-Fetch-Mode":["navigate"],"Cache-Control":["max-age=0"],"Sec-Ch-Ua":["\"Google Chrome\";v=\"111\", \"Not(A:Brand\";v=\"8\", \"Chromium\";v=\"111\""],"Sec-Ch-Ua-Platform":["\"Linux\""],"Upgrade-Insecure-Requests":["1"],"Sec-Fetch-User":["?1"],"Accept-Language":["en-US,en;q=0.9"],"User-Agent":["Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36"],"Sec-Fetch-Dest":["document"],"Cookie":[],"Sec-Ch-Ua-Mobile":["?0"],"Sec-Fetch-Site":["none"],"Accept-Encoding":["gzip, deflate, br"]},"tls":{"resumed":false,"version":772,"cipher_suite":4867,"proto":"h2","server_name":"netbox.fqdn"}},"user_id":"","duration":0.00596638,"size":0,"status":502,"resp_headers":{"Alt-Svc":["h3=\":443\"; ma=2592000"],"Server":["Caddy"]}}
netbox-docker-netbox-1               | django.db.migrations.exceptions.NodeNotFoundError: Migration dcim.0131_consoleport_speed depends on nonexistent node ('dcim', '0130_sitegroup'). Django tried to replace migration dcim.0130_sitegroup with any of [dcim.0003_squashed_0130] but wasn't able to because some of the replaced migrations are already applied.
netbox-docker-netbox-1               | [ Use DB_WAIT_DEBUG=1 in netbox.env to print full traceback for errors here ]
netbox-docker-netbox-1               | ⏳ Waiting on DB... (3s / 30s)
netbox-docker-netbox-1               | django.db.migrations.exceptions.NodeNotFoundError: Migration dcim.0131_consoleport_speed depends on nonexistent node ('dcim', '0130_sitegroup'). Django tried to replace migration dcim.0130_sitegroup with any of [dcim.0003_squashed_0130] but wasn't able to because some of the replaced migrations are already applied.
netbox-docker-netbox-1               | [ Use DB_WAIT_DEBUG=1 in netbox.env to print full traceback for errors here ]
netbox-docker-netbox-1               | ⏳ Waiting on DB... (6s / 30s)
netbox-docker-netbox-1               | django.db.migrations.exceptions.NodeNotFoundError: Migration dcim.0131_consoleport_speed depends on nonexistent node ('dcim', '0130_sitegroup'). Django tried to replace migration dcim.0130_sitegroup with any of [dcim.0003_squashed_0130] but wasn't able to because some of the replaced migrations are already applied.
netbox-docker-netbox-1               | [ Use DB_WAIT_DEBUG=1 in netbox.env to print full traceback for errors here ]
netbox-docker-netbox-1               | ⏳ Waiting on DB... (9s / 30s)
netbox-docker-netbox-1               | django.db.migrations.exceptions.NodeNotFoundError: Migration dcim.0131_consoleport_speed depends on nonexistent node ('dcim', '0130_sitegroup'). Django tried to replace migration dcim.0130_sitegroup with any of [dcim.0003_squashed_0130] but wasn't able to because some of the replaced migrations are already applied.
netbox-docker-netbox-1               | [ Use DB_WAIT_DEBUG=1 in netbox.env to print full traceback for errors here ]
netbox-docker-netbox-1               | ⏳ Waiting on DB... (12s / 30s)
```

### Content of docker-compose.override.yml

```yaml
version: '3.4'
services:
  netbox:
    environment:
      REMOTE_AUTH_ENABLED: "True"
      REMOTE_AUTH_BACKEND: "netbox.authentication.LDAPBackend"
      AUTH_LDAP_SERVER_URI: "ldaps://AD DC DN"
      AUTH_LDAP_BIND_DN: "CN=BINDACCOUNTCN"
      AUTH_LDAP_BIND_PASSWORD: "SECRETPASSWORD"
      AUTH_LDAP_USER_SEARCH_BASEDN: "OU=Users,USERS OU"
      AUTH_LDAP_GROUP_SEARCH_BASEDN: "OU=Groups,GROUP OU"
      AUTH_LDAP_REQUIRE_GROUP_DN: "CN=Netbox-Users,GROUP DN"
      AUTH_LDAP_IS_ADMIN_DN: "CN=Netbox-Users,GROUP CN"
      AUTH_LDAP_IS_SUPERUSER_DN: "CN=Netbox-Admins,GROUP CN"
      LDAP_IGNORE_CERT_ERRORS: "true"
      AUTH_LDAP_GROUP_TYPE: "NestedGroupOfNamesType"
```

**Comment by pts-jeffreyjewitt** [comment]
We are aware that the DB might be unrecoverable.
We're hoping that we can get any of the data from the DB
Thanks

**Comment by tobiasge** [comment]
Please read all release notes from Netbox. You can't directly upgrade to the newest version. You have to go to version 2.11.x first.

**Reply by pts-jeffreyjewitt** [reply]
Our issue is that we don't know the version we were on, and therefore don't know what version we can upgrade to.
We also have an issue where we can't start the app on any version of 2.x

**Reply by tobiasge** [reply]
You can find the version in the settings.py file. An update to 2.11.x will work from all older releases as far as I know.
You also need to checkout the correct version of netbox-docker. For Netbox 2.11.x this is our release 1.2.0.

**Reply by pts-jeffreyjewitt** [reply]
We don't have the original image, it wasn't tagged with a image number after a mess up a year ago, and the socker image was accidentally removed from the server.  That's our problem.

**Reply by tobiasge** [reply]
Ok, then just checkout version 1.2.0 of netbox-docker and try to start it. This should start all needed migrations to that version.
You can also try to only start the Postgres first, to see if the DB can be loaded. If it complains about an old database version the you have to set the version of the Postgres image to the older one used.

**Reply by githubpoints** [reply]
Generic error messages while trying to start netbox
    #image: netboxcommunity/netbox:v2.5.12-ldap
    #ModuleNotFoundError: No module named 'startup_script_utils'
    #image: netboxcommunity/netbox:v2.6.7-ldap
    #ModuleNotFoundError: No module named 'startup_script_utils'
    #image: netboxcommunity/netbox:v2.7.2-ldap
    #ImportError: cannot import name 'AdminGroup' from 'users.models'
    #image: netboxcommunity/netbox:v2.7.6-ldap
    #ImportError: cannot import name 'AdminGroup' from 'users.models'
    #image: netboxcommunity/netbox:v2.8.3-ldap
    #ImportError: cannot import name 'AdminGroup' from 'users.models' (/opt/netbox/netbox/users/models.py)
    #image: netboxcommunity/netbox:v2.9.11-ldap
    #ImportError: cannot import name 'Location' from 'dcim.models'
    #image: netboxcommunity/netbox:v2.10.6-ldap
    #ModuleNotFoundError: No module named 'django'
    #image: netboxcommunity/netbox:v2.11.12-ldap
    #ModuleNotFoundError: No module named 'django'
    #image: netboxcommunity/netbox:v2.5.8-ldap
    #ModuleNotFoundError: No module named 'startup_script_utils'
    image: netboxcommunity/netbox:v2.10.0-ldap
    #ModuleNotFoundError: No module named 'startup_script_utils'

It really feels like it tries to do migrations, then bombs out

netboxcommunity/netbox:v2.11.12-ldap
Traceback (most recent call last):
  File "./manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/core/management/__init__.py", line 363, in execute
    settings.INSTALLED_APPS
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/conf/__init__.py", line 82, in __getattr__
    self._setup(name)
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/conf/__init__.py", line 69, in _setup
    self._wrapped = Settings(settings_module)
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/conf/__init__.py", line 170, in __init__
    mod = importlib.import_module(self.SETTINGS_MODULE)
  File "/usr/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 848, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/opt/netbox/netbox/netbox/settings.py", line 46, in <module>
    from netbox import configuration
  File "/opt/netbox/netbox/netbox/configuration.py", line 71, in <module>
    _loaded_configurations = read_configurations(
  File "/opt/netbox/netbox/netbox/configuration.py", line 33, in read_configurations
    _import(f"{config_module}.{main_config}", main_config_path, loaded_configurations)
  File "/opt/netbox/netbox/netbox/configuration.py", line 20, in _import
    spec.loader.exec_module(module)
  File "/etc/netbox/config/configuration.py", line 20, in <module>
    def _read_secret(secret_name: str, default: str | None = None) -> str | None:
TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'
⏳ Waiting on DB... (18s / 30s)

Container then crashes and won't start.
DB appears to be up fine

PostgreSQL Database directory appears to contain a database; Skipping initialization

2023-04-28 13:33:15.109 UTC [1] LOG:  starting PostgreSQL 15.2 on x86_64-pc-linux-musl, compiled by gcc (Alpine 12.2.1_git20220924-r4) 12.2.1 20220924, 64-bit
2023-04-28 13:33:15.109 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
2023-04-28 13:33:15.109 UTC [1] LOG:  listening on IPv6 address "::", port 5432
2023-04-28 13:33:15.131 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2023-04-28 13:33:15.153 UTC [24] LOG:  database system was shut down at 2023-04-28 13:33:09 UTC
2023-04-28 13:33:15.188 UTC [1] LOG:  database system is ready to accept connections

**Comment by f0o** [comment]
Managed to upgrade from 2.6.x to release using this process:

1. run pg_dump on original 2.6.x installation
2. checkout tag `1.2.0` of netbox-docker and change the docker-compose.override.yml to include `image: netboxcommunity/netbox:v2.11.9` for netbox
3. `docker compose up -d postgres` to only start postgres
4. import the pg_dump from Step1.
5. `docker compose up` and wait for migrations to finish
6. get a new pg_dump from the postgres container and store it somewhere safe
7. `docker compose down` and delete all volumes (example: `docker volume rm $(docker volume ls -q)`)
8. checkout branch `release` and remove the modification from Step2.
9. `docker compose up -d postgres` to only start postgres (again)
10. import the pg_dump from Step6 (!)
11. `docker compose up` and wait for migrations to finish (again)
12. You're now on latest netbox with migrated data!

**Reply by FlorianHeigl** [reply]
will do the same since I have a safe pg_dump as a starting point. though the upgrade should have been neglible, being between versions less than a year apart.
my $0.02 would be
- if  you have a db dump, dont assume you'll have to lose any of your data. otherwise the data would not be in the database.
- it would be better if the docs didn't primarily recommend pulling the latest version, we should figure out an easy guideline for determining what one ought to check out (the wording 'if you instead want to update to a specific version, ...' is leading one to think that, no, why would i want something special there, i dont' need a specific version. you'll check the steps for 1.0.x, see those don't apply, move forward to the DB version upgrade steps and kaboom. Quite the nightmare still (though it was worse a few years ago)
- something is off with these "migration" scripts 

- bacula manages do flawless iterative updates on postgres since 2003 and, so like 21 years, and no db migration inconsistencies. simply not possible. i've also 'ops'ed on backends with millions of users and we had a complete informational disconnect from the devs, and there would have been scorched earth if those migration scripts weren't capable of finding out when and if they need to run. i hope whatever framework is behind the problem can some day handle that better. the only other software I saw that had as many problems was confluence, and they at least had safeguards and a version sticker on the DB + OS. and it's not like they were the masters of system integration.
